tee-worker/omni-executor/intent/executors/cross-chain/src/lib.rs

```diff
@@ -176,17 +208,269 @@ impl<
-				)?;
-				todo!("CrossChainSwap is not implemented yet");
+				// TODO: update this when we have more providers
+				let SingleChainSwapProvider::Pumpx(pumpx_config) = scsp;
+
+				let chain_id = match swap_order.to_asset {
+					ChainAsset::Ethereum(..) => ChainId::EVM,
+					ChainAsset::Solana(_) => ChainId::Solana,
+				};
+				let usd_worth = std::str::from_utf8(&pumpx_config.usd_worth)
+					.map_err(|_| {
+						log::error!("Failed to parse usd_worth");
+					})
+					.map(|v| v.to_string())?;
+				let token_ca = std::str::from_utf8(&pumpx_config.token_ca)
+					.map_err(|_| {
+						log::error!("Failed to parse token_ca");
+					})
+					.map(|v| v.to_string())?;
+				let Some(chain_type) = ChainType::from_pumpx_chain_id(chain_id.to_number() as u32)
+				else {
+					log::error!("Unsupported chain id: {:?}", chain_id);
+					return Err(());
+				};
+
+				let wallet_address = self
+					.pumpx_signer_client
+					.request_wallet(chain_type, pumpx_config.wallet_index, *account_id.as_ref())
+					.await
+					.map_err(|e| log::error!("Could not get wallet from pumpx-signer: {:?}", e))?;
+
+				let cross_order_data = CreateCrossOrderData {
+					request_id: intent_id,
+					chain_id: chain_id.clone(),
+					info: vec![CrossOrderInfo {
+						chain_id: chain_id.clone(),
+						wallet_index: pumpx_config.wallet_index,
+						address: wallet_address.to_hex(),
+						amount: from_amount_string.clone(),
+						usd: usd_worth,
+						token_ca: token_ca.clone(),
+					}],
+				};
+				let storage = PumpxJwtStorage::new(self.storage_db.clone());
+				let Some(access_token) = storage.get(&(account_id.clone(), AUTH_TOKEN_ACCESS_TYPE))
+				else {
+					log::error!("Failed to get access token from storage");
+					return Err(());
+				};
+
+				self.pumpx_api
+					.create_cross_order(&access_token, cross_order_data)
+					.await
+					.map_err(|_| {
+						log::error!("Failed to create cross order");
+					})?;
+
+				let pumpx_order_response: Option<Vec<u8>>;
+
+				if swap_order.from_asset.is_same_chain(&swap_order.to_asset) {
+					let user_trade_info =
+						self.pumpx_api.get_user_trade_info(&access_token).await.map_err(|_| {
+							log::error!("Failed to get user trade info");
+						})?;
+
+					let order_response = match pumpx_config.order_type {
+						PumpxOrderType::Market => {
+							let new_market_order = NewMarketOrder {
+								request_id: intent_id,
+								chain_id: chain_id.clone(),
+								token_ca,
+								swap_type: match pumpx_config.swap_type {
+									1 => SwapType::Buy,
+									2 => SwapType::Sell,
+									_ => {
+										log::error!(
+											"Unsupported swap type: {}",
+											pumpx_config.swap_type
+										);
+										return Err(());
+									},
+								},
+								amount_in: from_amount_string.clone(),
+								double_out: pumpx_config.double_out,
+								is_one_click: pumpx_config.is_one_click,
+								address: "todo: what's this??".to_string(),
```

Comment:
This is the wallet address of tx sender - in this case the omni-account address on this chain, which can be retrieved from signer by `omni-account` + `chain_id` + `wallet_index`

Reply:
ah I forgot about this one, I only changed it in the other order type 👍🏼

Reply:
done